File: Prod.RutaDigital.Presentacion.Angular/ClientApp/src/app/interfaces/premio.ts
import { SafeUrl } from "@angular/platform-browser";

export interface PremioPublicidadResponse {
    // foto: string;
    id_publicidad_premio?: number;
    imagen?: string;
    numArray?: Blob;
    imagenPremioPublicidad?: SafeUrl;
}

export interface PremioTipoResponse {
    // foto: string;
    id_tipo_premio?: number;
    descripcion?: string;
    imagen?: string;
    numArray?: Blob;
    imagenTipoPremio?: SafeUrl;
}
  
export interface PremioResponse {
    id_premio?: number;
    id_tipo_premio?: number;
    tipo_premio? : string;
    id_nivel_madurez?: number;
    nivel_madurez?: string;
    id_tipo_canje?: number;
    tipo_canje?: string;
    foto?: string;
    nombre?: string;
    descripcion_corta?: string;
    detalle_premio?: string;
    puntos_produce? : number;
    correo? : string;
    numArray?: Blob;
    imagenPremio?: SafeUrl;
}
  

File: Prod.RutaDigital.Entidades/Premio/Premio.cs
﻿namespace Prod.RutaDigital.Entidades;

public class Premio
{
    public int id_premio { get; set; }
    public int id_tipo_premio { get; set; }
    public string? tipo_premio { get; set; }
    public int id_nivel_madurez { get; set; }
    public int id_tipo_canje { get; set; }
    public string? foto { get; set; }
    public string? nombre { get; set; }
    public string? descripcion_corta { get; set; }
    public string? detalle_premio { get; set; }
    public int puntos_produce { get; set; }
    public string? correo { get; set; }
}